Ginger/GingerCoreNET/RunLib/DynamicExecutionLib/DynamicExecutionManager.cs

```diff
@@ -369,24 +371,585 @@ public static void CreateRealRunSetFromDynamic(RunsetExecutor runsetExecutor, Ad
+                        }
+                        if (runsetOperationConfig.RunAt != null)
+                        {
+                            runSetOperation.RunAt = (RunSetActionBase.eRunAt)Enum.Parse(typeof(RunSetActionBase.eRunAt), runsetOperationConfig.RunAt.ToString(), true);
+                        }
+
+                        if (!dynamicRunsetConfigs.Exist)
+                        {
+                            runSetConfig.RunSetActions.Add(runSetOperation);
+                        }
+                    }
+                }
+            }
+
+            // Set config
+            runsetExecutor.RunSetConfig = runSetConfig;
+        }
+
+        private static T FindItemByIDAndName<T>(Tuple<string, Guid?> id, Tuple<string, string> name, ObservableList<T> repoLibrary)
+        {
+            T item = default(T);
+
+            try
+            {
+                if (id.Item2 != null && id.Item2 != Guid.Empty)
+                {
+                    if (typeof(T).GetProperty(id.Item1) != null)
+                    {
+                        item = repoLibrary.Where(x => (Guid)(typeof(T).GetProperty(id.Item1).GetValue(x)) == id.Item2).FirstOrDefault();
+                    }
+                    else if (typeof(T).GetField(id.Item1) != null)
+                    {
+                        item = repoLibrary.Where(x => (Guid)(typeof(T).GetField(id.Item1).GetValue(x)) == id.Item2).FirstOrDefault();
+                    }
+                }
+
+                if (item == null && !string.IsNullOrEmpty(name.Item2))
+                {
+                    if (typeof(T).GetProperty(name.Item1) != null)
+                    {
+                        item = repoLibrary.Where(x => typeof(T).GetProperty(name.Item1).GetValue(x).ToString().ToLower() == name.Item2.ToLower()).FirstOrDefault();
+                    }
+                    else if (typeof(T).GetField(name.Item1) != null)
+                    {
+                        item = repoLibrary.Where(x => typeof(T).GetField(name.Item1).GetValue(x).ToString().ToLower() == name.Item2.ToLower()).FirstOrDefault();
+                    }
+                }
+
+                if (item != null)
+                {
+                    return item;
+                }
+                else
+                {                    
+                    string error = string.Format("Failed to find {0} with the details '{0}/{1}'", typeof(T), name.Item2.ToLower(), id.Item2);
+                    Reporter.ToLog(eLogLevel.ERROR, error);
+                    throw new Exception(error);
+                }
+            }
+            catch (Exception ex)
+            {
+                string error = string.Format("Failed to find {0} with the details '{0}/{1}'", typeof(T), name.Item2.ToLower(), id.Item2);
+                Reporter.ToLog(eLogLevel.ERROR, error, ex);
+                throw new Exception(error, ex);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['System.Exception' should not be thrown by user code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=4812633)